core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/AvroUpsertMonitor.scala

```diff
@@ -256,7 +256,12 @@ class AvroUpsertMonitorActor(
           publishMessageToUpdateImportStatus(attributes.importId, Option(status), ImportStatuses.Upserting, None)
           toFutureTry(initUpsert(attributes.upsertFile, attributes.importId, message.ackId, workspace, attributes.userEmail)) map {
             case Success(importUpsertResults) =>
-              val msg = s"Successfully updated ${importUpsertResults.successes} entities; ${importUpsertResults.failures.size} updates failed."
+              val failureMessages = stringMessageFromFailures(importUpsertResults.failures, 100)
+              val baseMsg = s"Successfully updated ${importUpsertResults.successes} entities; ${importUpsertResults.failures.size} updates failed."
+              val msg = if (importUpsertResults.failures.isEmpty)
+                baseMsg
+              else
+                baseMsg + s" First 100 failures are: $failureMessages"
               publishMessageToUpdateImportStatus(attributes.importId, Option(status), ImportStatuses.Done, Option(msg))
```

Comment:
this is where we actually change the API response to the end user. Rawls sends this message to Import Service to set the job's status and message; the API reads the status+message from Import Service. By changing the message here, we change what the user will see.